invenio_app_ils/patrons/views.py

```diff
@@ -0,0 +1,87 @@
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio App ILS Patrons views."""
+
+from flask import Blueprint
+from flask import abort, current_app
+from flask_login import current_user
+from invenio_circulation.search.api import search_by_patron_item_or_document
+from invenio_app_ils.circulation.search import get_loans_aggregated_by_states
+from elasticsearch import VERSION as ES_VERSION
+from invenio_app_ils.permissions import need_permissions
+from webargs import fields
+from webargs.flaskparser import use_args
+lt_es7 = ES_VERSION[0] < 7
+
+
+def get_user_loan_information_blueprint(app):
+    """Add patron views to the blueprint."""
+    blueprint = Blueprint("invenio_app_ils_patrons", __name__,)
+
+    @blueprint.route('/me/loans')
+    @use_args({"document_pid": fields.Str(required=True)})
+    def get_user_information(args):
+        return retrieve_user_information(args["document_pid"])
+    return blueprint
+
+
+def retrieve_user_loans_information(patron_pid, document_pid):
+    """Updates the information about the loans of a concrete document of the currently logged in patron."""
+    active_requested_loan_states = (
+        current_app.config["CIRCULATION_STATES_LOAN_REQUEST"]
+        + current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"]
+    )
+    past_loan_states = (
+        current_app.config["CIRCULATION_STATES_LOAN_COMPLETED"]
+    )
+
+    user_information = {
+        "user_has_request": False,
+        "user_has_loan": False,
+        "user_last_loan": None,
+    }
+
+    loans_search = get_loans_aggregated_by_states(document_pid, active_requested_loan_states+past_loan_states, patron_pid)
+    # No need for the loan hits
+    loans_search = loans_search[:0]
+    loan_result = loans_search.execute()
+    for bucket in loan_result.aggregations.states.buckets:
+        if bucket["key"] in current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"] \
+                and bucket["doc_count"] > 0:
+            user_information['user_has_loan'] = True
+        elif bucket["key"] in current_app.config["CIRCULATION_STATES_LOAN_REQUEST"] \
+                and bucket["doc_count"] > 0:
+            user_information['user_has_request'] = True
+        elif bucket["key"] in current_app.config["CIRCULATION_STATES_LOAN_COMPLETED"] \
+                and bucket["doc_count"] > 0:
+            search = search_by_patron_item_or_document(
+                patron_pid=patron_pid, document_pid=document_pid, filter_states=past_loan_states
+            )
+
+            search = search.sort({"end_date": {"order": "desc"}})
+            search_result = search.execute()
+            has_past_loans = search_result.hits.total > 0 if lt_es7 else search_result.hits.total.value > 0
+            if has_past_loans:
+                last_loan = search_result.hits[0]
+                user_information['user_last_loan'] = last_loan['end_date']
+
+    return user_information
+
+
+@need_permissions("patron-loans-information")
```

Comment:
```suggestion
@need_permissions("patron-loans")
```

Can you move this above? Actually I would change it:
```
    @blueprint.route('/me/loans')
    @use_args({"document_pid": fields.Str(required=True)})
    @need_permissions("patron-loans")
    def get_user_information(args):
        patron_pid = str(current_user.id)
        return retrieve_user_information(patron_pid, args["document_pid"])
```

and:
```
def retrieve_user_information(patron_pid, document_pid):
```
so that this function can be called via command line, without a request context. We should try to keep httpe/request data in the views and not in the "controller" functions.